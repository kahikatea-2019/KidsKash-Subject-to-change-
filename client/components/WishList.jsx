import React from 'react'
import { connect } from 'react-redux'

// import component
import Wish from './Wish'

class WishList extends React.Component {
  constructor (props) {
    super(props)

    this.retrieveChildWishes = this.retrieveChildWishes.bind(this)
  }

  retrieveChildWishes () {
    const { wishes, currentUser } = this.props
    const childWishes = wishes.map(wish => {
      if (wish.child_id === currentUser) {
        return <Wish key={wish.id} wish={wish} />
      }
    })
    return childWishes
  }

  render () {
    const childWishes = this.retrieveChildWishes()
    return (
      <div className='wishlist'>
        <h2>Wishes</h2>
        {childWishes}
      </div>
    )
  }
}

function mapStateToProps (state) {
  return {
    wishes: state.wishes,
    currentUser: state.navigate.currentUser
  }
}

export default connect(mapStateToProps)(WishList)
