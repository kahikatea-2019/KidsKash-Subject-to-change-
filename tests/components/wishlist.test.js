import React from 'react'
import { shallow } from 'enzyme'
import WishList from '../../client/components/WishList'

const wishList = shallow(<WishList />)

test('test <WishList/> is set up correctly', () => {
  expect(true).toBe(true)
})

test('<WishList/> has 1 div', () => {
  expect(wishList.find('div').length).toBe(1)
})

test('<WishList/>', () => {
  expect(wishList.find('h2').text()).toBe('Mum/Dad: These are what i want :')
})

describe('WishList', () => {
  it('returns correctly', () => {
    expect(wishList).toMatchSnapshot()
  })
})
