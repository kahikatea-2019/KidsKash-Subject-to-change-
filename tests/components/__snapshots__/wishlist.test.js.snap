// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WishList returns correctly 1`] = `ShallowWrapper {}`;
